azerothcore/azerothcore-wotlk
# fix(Core/Spell): Prevent passive spells to proc Omen of Clarity

## Changes Proposed:

Today it procs due to the passive spell of shapeshifting. 
> Dire Bear Form (Passive) (Passive) [9635]

After testing, it also procs on passive talents. 

I propose we prevent it to proc from any passive spell source, but this needs thorough verification

-  Prevent passive spells to proc Omen of Clarity 

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9467
- Closes chromiecraft/chromiecraft#2455

## SOURCE:


N/A - We need to verify this doesn't break some other spell/talent, see discussion below

## Tests Performed:


- Tested ingame
- Build passed

> ========== Build: 3 succeeded, 0 failed, 16 up-to-date, 1 skipped ==========


## How to Test the Changes:

Do below steps repeatly to make sure it never procs.

1. Create druid
2. `.learn all my spells`
3. Use any shapeshift form
4. `/cancelform`


## Known Issues and TODO List:


- I'm unsure if this will cause problems with other spells and/or talents, please contribute to the discussion below if you have any source or thought..


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
